src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -1250,6 +1251,23 @@ public ChainedHeaderBlock GetBlockData(uint256 blockHash)
             return chainedHeaderBlock;
         }
 
+        /// <inheritdoc />
+        public ChainedHeaderBlock[] GetBlockData(List<uint256> blockHashes)
+        {
+            Dictionary<uint256, ChainedHeaderBlock> chainedHeaderBlocks;
+
+            lock (this.peerLock)
+            {
+                chainedHeaderBlocks = this.chainedHeaderTree.GetChainedHeaderBlocks(blockHashes).ToDictionary(chb => chb.ChainedHeader.HashBlock, chb => chb);
+            }
+
+            List<uint256> getFromStore = chainedHeaderBlocks.Where(kv => kv.Value != null && kv.Value.Block == null).Select(kv => kv.Key).ToList();
+            foreach ((Block block, int index) in this.blockStore.GetBlocks(getFromStore).Select((b, n) => (b, n)))
+                chainedHeaderBlocks[getFromStore[index]] = new ChainedHeaderBlock(block, chainedHeaderBlocks[getFromStore[index]].ChainedHeader);
+
+            return blockHashes.Select(h => chainedHeaderBlocks[h]).ToArray();
```

Comment:
Is there still a possibility that we can return an array of `ChainHeaderBlock`s that has a null block?

The consumer will need to check for this.

Reply:
We are mimicking the behavior of the single-block `GetBlockData`.

> The consumer will need to check for this.

Yes that is true for both the sinlgle- and multi-block versions of this method.